tests/installation/yast2_firstboot.pm

```diff
@@ -46,45 +47,41 @@ sub license {
     send_key $cmd{next};
 }
 
-sub welcome {
+sub firstboot_welcome {
     assert_screen 'welcome';
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
-sub clock_and_timezone {
+sub firstboot_timezone {
     assert_screen 'inst-timezone';
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
-sub user_setup {
-    my $is_not_shared_passwd = shift;
+sub firstboot_user {
     assert_screen 'local_user';
     enter_userinfo(username => get_var('YAST2_FIRSTBOOT_USERNAME'));
-    if (defined($is_not_shared_passwd)) {
-        send_key 'alt-t' if (is_opensuse);
-    }
-    else {
-        send_key 'alt-t' if (is_sle('>=12-sp4'));
-    }
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
     await_password_check;
-    wait_screen_change(sub { send_key $cmd{next}; }, 7) unless defined($is_not_shared_passwd);
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
-sub root_setup {
+sub firstboot_root {
     assert_screen 'root_user', 60;
     enter_rootinfo;
     wait_screen_change(sub { send_key $cmd{next}; }, 7);
 }
 
+sub firstboot_hostname {
+    assert_screen 'hostname';
+    wait_screen_change(sub { send_key $cmd{next}; }, 7);
+}
+
 sub run {
-    my $self = shift;
-    language_and_keyboard;
-    welcome;
-    $self->license;
-    clock_and_timezone;
-    user_setup(1);
-    root_setup;
+    my $self      = shift;
+    my $test_data = get_test_suite_data();
+    foreach my $client (@{$test_data->{clients}}) {
+        $self->$client;
```

Comment:
In general, we should avoid symbolic references as they cause all kinds of problems (typos, renaming, etc.). Better practice is to use a dispatch table:
```
my %actions = ( foo => \&foo,
                bar => \&bar,
                baz => sub { print 'baz!' } 
                ... 
              );
$actions{$action}->();
```
In that way you can also ensure that the name of the method from test data exists in the file.

Reply:
To be honest, I don't really understand the problem, nor the proposed solution. So I tried something, but er, not sure you will like it either (nor that it's even working)